src/mongocrypt-ctx-encrypt.c

```diff
@@ -938,6 +938,75 @@ generate_delete_tokens (_mongocrypt_crypto_t *crypto,
    return out;
 }
 
+static bool
+_check_for_payload_requiring_encryptionInformation (void *ctx,
+                                                    _mongocrypt_buffer_t *in,
+                                                    mongocrypt_status_t *status)
+{
+   bool *out = (bool *) ctx;
+
+   if (in->len < 1) {
+      CLIENT_ERR ("unexpected empty FLE payload");
+      return false;
+   }
+
+   if (in->data[0] == MC_SUBTYPE_FLE2InsertUpdatePayload) {
+      *out = true;
+      return true;
+   }
+
+   if (in->data[0] == MC_SUBTYPE_FLE2FindEqualityPayload) {
+      *out = true;
+      return true;
+   }
+
+   return true;
+}
+
+static bool
+must_omit_encryptionInformation (const char *command_name,
```

Comment:
This naming spooks me, because the return value is not actually the result value, which are both `bool`, yet the name implies the result is related to what it computes. I'm not sure what a better name might be, though. Maybe return a `struct` with two named values instead? Or maybe an enum of "yes, no, error" to be checked at the call site?

Reply:
I like the idea of returning a struct. Done.